lib/rollup-build.js

```diff
@@ -0,0 +1,46 @@
+const { resolve } = require('path')
+const { rollup } = require('rollup')
+
+/**
+ * Metalsmith Rollup plugin
+ */
+const plugin = (config) => async (files, metalsmith, done) => {
+  const input = resolve(config.input)
+
+  // Add Rollup plugin to read from Metalsmith files (not disk)
+  for (const file of Object.keys(files)) {
```

Comment:
Can we avoid this loop? I've had a look using [perf hooks](https://nodejs.org/api/perf_hooks.html), and this loop takes around 1ms, whereas something (very messy) like:

```
const file = Object.keys(files).filter(file => input.endsWith(file))?.[0]
if (file) {
  config.plugins.unshift(fromFiles(input, files[file].contents.toString()))
}
```

takes around 0.05ms.

Not a super noticeable performance enhancement though, so no biggie if not.

Reply:
Thanks @domoscargin

Yeah I didn't find it too clear [in the original plugin](https://github.com/attila/metalsmith-rollup/blob/master/lib/index.js#L33) but I suppose `metalsmith.source()` is important should `.endsWith()` match but the source file came from `node_modules` for example:

**Metalsmith file `common/example.js`** matches these two:
```
/path/to/project/node_modules/govuk-frontend/govuk/common/example.js
/path/to/project/src/common/example.js
```

If I take your idea but keep `metalsmith.source()`?

```mjs
const [path] = Object.keys(files).filter((path) =>
  resolve(metalsmith.source(), path) === config.input)
```

Reply:
Ah yeah, good point! A quick glance shows that's a fairly similar speed (0.88ms). I hadn't clocked the situation with almost-matching files. Seems like resolving the source is the main "slowdown" here, so I really don't think it's worth changing now!